packages/web/src/components/result/addons/Results.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { getClassName, isEqual } from '@appbaseio/reactivecore/lib/utils/helper';
+import types from '@appbaseio/reactivecore/lib/utils/types';
+
+class Results extends React.Component {
+	shouldComponentUpdate(nextProps) {
+		/*
+			Use shouldComponentUpdate because PureComponent uses shallow comparison of objects.
+			If props contain complex data structures, it may produce false-negatives for
+			deeper differences.
+		*/
+		return !isEqual(nextProps, this.props);
+	}
+
+	render() {
+		if (this.props.hasCustomRender) {
+			return this.props.getComponent();
+		}
+
+		return (
+			<div
+				className={`${this.props.listClass} ${getClassName(this.props.innerClass, 'list')}`}
+			>
+				{this.props.filteredResults.map((item, index) =>
+					this.props.renderItem(item, () => {
+						this.props.triggerClickAnalytics(this.props.base + index);
+					}),
+				)}
+			</div>
+		);
+	}
+}
+
+Results.propTypes = {
+	hasCustomRender: types.boolRequired,
+	innerClass: types.style,
+	renderItem: types.func,
+	base: types.number,
+	getComponent: types.func,
+	listClass: types.string,
+	triggerClickAnalytics: types.func,
+};
```

Comment:
@jyash97 `filteredResults` is not added in props validations, don't you have eslint running?
I'm not sure how we can ignore the eslint errors, we already set a pre-commit hook to avoid the linting errors.
We shouldn't bypass it.

Reply:
Checking